### PR TITLE
Update UI to use v2 patient endpoint

### PR DESCRIPTION
**Pull request by BrianMaki**
# Implements [AB#14271](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/14271)

## Description

- Refactored name and location of ApiPatientException to ApiException
- Refactored location of ApiProblemDetails
- Updated services, delegates and units based on refactoring
- Updated UI to use v2 patient endpoint
- JSON result will only return resource payload and warning if they are not null.  If waning object is null, then it will not be included in the JSON.  Same goes for resource payload.
- Rider format changes applied on some files that had not been previously affected

**200:**

Screenshot 2022-11-10 at 3 56 56 PM

**404:**

Screenshot 2022-11-10 at 2 39 42 PM

**500:**

Screenshot 2022-11-10 at 3 40 55 PM

**Warning:**

Screenshot 2022-11-10 at 3 44 40 PM

## Testing

- [x] Unit Tests Updated
- [ ] Functional Tests Updated
- [ ] Not Required

## UI Changes

No but action, service and model were updated.

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

### CODE REVIEW

File: Apps/CommonData/src/ErrorHandling/ApiException.cs [hunk 1]
```diff
@@ -13,69 +13,70 @@
...
         /// <param name="instance">The instance associated with the exception.</param>
         /// <param name="statusCode">The http status code associated with the exception.</param>
-        public ApiPatientException(string detail, string instance, HttpStatusCode statusCode)
+        public ApiException(string detail, string instance, HttpStatusCode statusCode)
         {
-            this.ProblemType = "api-patient-exception";
+            this.ProblemType = "api-exception";
             this.Detail = detail;
-            this.Title = "Api Patient Service Exception";
+            this.Title = "Custom Exception Handling";
```
Comment (sslaws) [inline Apps/CommonData/src/ErrorHandling/ApiException.cs:39]: I think the title should be something like Health Gateway Exception

File: Apps/WebClient/src/ClientApp/src/store/modules/user/actions.ts [hunk 2]
```diff
@@ -306,19 +305,9 @@ export const actions: UserActions = {
                                 resolve();
                             });
                     } else {
-                        if (result.resultError?.statusCode === 429) {
```
Comment (sslaws) [inline Apps/WebClient/src/ClientApp/src/store/modules/user/actions.ts:309]: This code doesn't look right but don't we need 429 logic still?

**Review by sslaws — CHANGES_REQUESTED** [review]

**Review by MikeLyttle — APPROVED** [review]

File: Apps/CommonData/src/ErrorHandling/ApiException.cs (hunk 1 above)
Comment (BrianMaki) [reply Apps/CommonData/src/ErrorHandling/ApiException.cs:39]: Changed.

File: Apps/WebClient/src/ClientApp/src/store/modules/user/actions.ts (hunk 2 above)
Comment (BrianMaki) [reply Apps/WebClient/src/ClientApp/src/store/modules/user/actions.ts:309]: @sslaws  Spoke with @MikeLyttle  about this.  That line of code was erroneous.  The 429 is handled in the catch block on line 314.

**Review by sslaws — APPROVED** [review]

**Pull request merged**